doc/conditional_independence.rst

```diff
@@ -0,0 +1,180 @@
+It is because of this reason, one would typically like the most powerful test given assumptions about
+the data. With that in mind, there are various approaches to CI testing that are typically more powerful
+with certain assumptions on the underlying data distribution.
+
+Conditional Mutual Information
+------------------------------
+Conditional mutual information (CMI) is a general formulation of CI, where CMI is defined as
+:math::
+    
+    \\int log \frac{p(x, y | z)}{p(x | z) p(y | z)}
+
+As we can see, CMI is equal to 0, if and only if :math:`p(x, y | z) = p(x | z) p(y | z)`, which
+is exactly the definition of CI. CMI is completely non-parametric and thus requires no assumptions
+on the underlying distributions. Unfortunately, CMI is notoriously difficult to estimate. There are
+various proposals in the literature for estimating CMI, which we summarize here:
+
+- The Kraskov, Stogbauer and Grassberger (KSG) estimate approach estimates mutual information
+  via k-NN statistics :footcite:`kraskov_estimating_2004`. It was generalized to CMI by :footcite:`frenzel_partial_2007`.
+  This class of estimators is asymptotically correct, meaning if we had an infinite amount of data
+  we would obtain the true value of the CMI. However, it relies on statistics generated from the k-NN,
+  which if we implement the naive approach using a KDTree, then it generally suffers in high-dimensions.
+  In our examples, we see it suffer with dimensionality > 4 or 5.
+
+  Estimates of CMI can be converted into a CI hypothesis test by permutation testing :footcite:`Runge2018cmi`.
+  One can generate estimated samples from the null distribution by permuting samples in an intelligent manner
+  and then the CMI value generated from the actual observed data can be compared to the CMI values computed
+  on the permutated datasets to estimate a pvalue.
+
+  It is worth noting that if one has "robust" estimates of k-NN using for example a model that
+  is effective in high-dimensions, then the KSG estimator for CMI may be quite good. For example,
+  one can use variants of Random Forests to generate adaptive nearest-neighbor estimates in high-dimensions
+  or on manifolds, such that the KSG estimator is still quite good.
+
+.. autosummary::
+   :toctree: generated/
+
+    CMITest
+
+- The Classifier Divergence approach estimates CMI using a classification model.
+
+.. autosummary::
+   :toctree: generated/
+
+    ClassifierCMITest
+
+- Direct posterior estimates can be implemented with a classification model by directly
+  estimating :math:`P(y|x)` and :math:`P(y|x,z)`, which can be used as plug-in estimates
+  to the equation for CMI.
+
+Partial (Pearson) Correlation
+-----------------------------
+Partial correlation based on the Pearson correlation is equivalent to CMI in the setting
+of normally distributed data. Computing partial correlation is fast and efficient and
+thus attractive to use. However, this **relies on the assumption of multivariate Gaussianity**,
```

Comment:
"Gaussianity"?

Reply:
Seems like it is a word? https://en.wikipedia.org/wiki/Non-Gaussianity hehe. Any suggestions?

Reply:
I think Gaussianity is quite common, but the combination with "multivariate" sounds odd. What about "However, this relies on the assumption that the variables are Gaussian."

Reply:
Modified